### ISSUE TITLE
Include minimum and recommended server specs in README

### ISSUE THREAD

**Issue by verandaguy** [nylas/sync-engine#397]
Currently, attempting to `vagrant up` on smaller instances results in VirtualBox entering the invalid `gurumeditation` state. Often, this is related to VirtualBox not being able to allocate enough RAM to launch a machine.

It would be useful for information relating to minimum system specs to be posted in README.md alongside the installation instructions.

Steps to reproduce:
- Launch a Debian 8.5 "Jessie" instance via the DigitalOcean cloud hosting provider
- Use the following instructions to install VirtualBox: https://wiki.debian.org/VirtualBox#Debian_8_.22Jessie.22
- `apt update`
- `apt install git`
- `git clone https://github.com/nylas/sync-engine.git`
- `cd sync-engine`
- `vagrant up`

Actual output:

```
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Checking if box 'ubuntu/precise64' is up to date...
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
    default: Adapter 2: hostonly
==> default: Forwarding ports...
    default: 5000 => 5000 (adapter 1)
    default: 5001 => 5001 (adapter 1)
    default: 5002 => 5002 (adapter 1)
    default: 5003 => 5003 (adapter 1)
    default: 5004 => 5004 (adapter 1)
    default: 5005 => 5005 (adapter 1)
    default: 5006 => 5006 (adapter 1)
    default: 5007 => 5007 (adapter 1)
    default: 5008 => 5008 (adapter 1)
    default: 5009 => 5009 (adapter 1)
    default: 8000 => 8000 (adapter 1)
    default: 5555 => 5555 (adapter 1)
    default: 9200 => 9200 (adapter 1)
    default: 22 => 2222 (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
The guest machine entered an invalid state while waiting for it 
to boot. Valid states are 'starting, running'. The machine is in the 
'gurumeditation' state. Please verify everything is configured properly and 
try again.
```

I have tried this with instances which have up to 4GB of RAM.

**Comment by grinich** [comment]
Hey @verandaguy -- it should totally work on an instance with <1 GB of RAM.

**Comment by verandaguy** [comment]
@grinich I'm afraid that's just not true. For kicks, I just tried to `vagrant up` on an identically-specced Debian 8.7 machine (500MB RAM, 20GB storage, 1vCPU) on DigitalOcean.

I installed Vagrant and VirtualBox, and attempted to launch the VM. After about a minute, I got the same error as the one I described above.

I can't commit time to run more thorough tests and add data points since I no longer use Nylas, but I thought I'd update this thread since it still seems to be an active issue.
